crates/next-core/js/src/entry/config/next.js

```diff
@@ -3,8 +3,16 @@ import { PHASE_DEVELOPMENT_SERVER } from "next/dist/shared/lib/constants";
 
 const loadNextConfig = async () => {
   const nextConfig = await loadConfig(PHASE_DEVELOPMENT_SERVER, process.cwd());
-  nextConfig.rewrites = await nextConfig.rewrites?.();
+
+  nextConfig.generateBuildId = await nextConfig.generateBuildId?.();
+  nextConfig.headers = await nextConfig.headers?.();
```

Comment:
the `loadCustomRoutes` from `next/dist/lib/load-custom-routes` will perform some validation and loading for `headers`/`redirects`/`rewrites`.